pandas/_libs/tslibs/offsets.pyx

```diff
@@ -620,6 +628,8 @@ class _BaseOffset:
         TimeStamp
             Rolled timestamp if not on offset, otherwise unchanged timestamp.
         """
+        # TODO: try to avoid runtime/circular import
+        from pandas import Timestamp
```

Comment:
can you not cimport this?

Reply:
we need Timestamp (not cimport-able), not _Timestamp (cimport-able) here.  We could re-instate a `as_timestamp` func to only do the runtime import in one place, and after a (fast) isinstance check.

I asked on the cython mailing list [yesterday](https://groups.google.com/forum/#!topic/cython-users/ikkDWc1bXkU) about handling of circular cimports.  It seems pretty hit-and-miss.